tests/grid/serialization_01.cc

```diff
@@ -0,0 +1,122 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2008 - 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// Test serialization with triangulations.
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/lac/la_vector.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+#include <boost/archive/text_iarchive.hpp>
+#include <boost/archive/text_oarchive.hpp>
+
+#include "./tests.h"
+
+using namespace dealii;
+
+template <int dim>
+class InterpolationFunction : public Function<dim>
+{
+public:
+  InterpolationFunction()
+    : Function<dim>(1)
+  {}
+
+  virtual double
+  value(const Point<dim> &p, const unsigned int component = 0) const
+  {
+    return p.norm();
+  }
+};
+
+template <int dim, typename TriangulationType>
+void
+test(TriangulationType &triangulation)
+{
+  DoFHandler<dim> dof_handler(triangulation);
+  dof_handler.distribute_dofs(FE_Q<dim>(2));
+
+  using VectorType = Vector<double>;
+
+  VectorType vector(dof_handler.n_dofs());
+
+  VectorTools::interpolate(dof_handler, InterpolationFunction<dim>(), vector);
+
+  VectorType vector_loaded(dof_handler.n_dofs());
+
+  print_statistics(triangulation, false);
+
+  std::stringstream stream;
+  {
+    boost::archive::text_oarchive oa(stream);
+    oa << triangulation;
+    oa << vector;
```

Comment:
I don't think we need to test here the vector. There should be other tests whether the serialization/deserialization of vectors work.